test/integration/consul-container/test/envoy_extensions/testdata/wasm_test_files/wasm_add_header.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func main() {
+	proxywasm.SetVMContext(&vmContext{})
+}
+
+type vmContext struct {
+	// Embed the default VM context here,
+	// so that we don't need to reimplement all the methods.
+	types.DefaultVMContext
+}
+
+func (*vmContext) NewPluginContext(contextID uint32) types.PluginContext {
+	return &pluginContext{}
+}
+
+type pluginContext struct {
+	// Embed the default plugin context here,
+	// so that we don't need to reimplement all the methods.
+	types.DefaultPluginContext
+}
+
+func (p *pluginContext) NewHttpContext(contextID uint32) types.HttpContext {
+	return &httpHeaders{}
+}
+
+type httpHeaders struct {
+	// Embed the default http context here,
+	// so that we don't need to reimplement all the methods.
+	types.DefaultHttpContext
+}
+
+func (ctx *httpHeaders) OnHttpResponseHeaders(int, bool) types.Action {
+	proxywasm.LogDebug("adding header: x-test:true")
+
+	err := proxywasm.AddHttpResponseHeader("x-test", "true")
+	if err != nil {
+		proxywasm.LogCriticalf("failed to add test header to response: %v", err)
```

Comment:
Should we abort the request here so that the test fails outright? I feel like this is unlikely to happen, but if it does, will it be hard to track down what is happening?

:thinking: I guess if the header isn't added then the test will fail outright anyways and we'd look in the proxy log and see this error. So I think this is probably OK as is, but I'll leave this here for your consideration.

Reply:
Probably not a bad call. I'll add something a little easier to spot in case this fails.

Reply:
Not seeing anything that I love as far as options. We can send an HTTP response with a 5xx code then check the status code in the tests. Or we can close the connection and that would cause an error but seems sort of like the wrong thing as well. What do you think?

Here are the methods available in `proxywasm` https://pkg.go.dev/github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm

Reply:
Let's just leave it as is then. I think I've convinced myself that in the rare case that the Wasm file can't add the header, there is still a failure and a trail of breadcrumbs to follow.